### DISCUSSION TITLE
Set current state for story

### DISCUSSION THREAD

**Discussion by steveoh** [SimeonC/storybook-xstate-addon#50] [Q&A]
Is there a way to set the state of the machine without going through the events? I have a machine that once it starts it moves its way through all the states and I'd like to have stories be fixed to a specific state.

**Comment by SimeonC** [comment]
That's a great idea! I think it should be possible if we stop the machine once it reaches a particular state. I haven't tried that so it may just break the component - worth a try!

It might be possible to just rehydrate the machine at a particular state and then stop the machine again too.

**Reply by steveoh** [reply]
My issue is that I have a service that auto moves my machine through state with callbacks and I can't pause in a state. I decoupled my machine from my component so I could mock the state object for now. But I wouldn't have to do that if I could test in a specific state.

**Reply by SimeonC** [reply]
In this addon via the XState dev tools we have access to the service/actor that's the third array value returned from `const [state, send, actor] = useMachine(...` or the direct value returned from `useInterpret`. In your use case if you do `actor.onTransition(state => { if (state.matches(...)) actor.stop() }` does that "pause" the machine correctly?

**Reply by steveoh** [reply]
Unsure. I updated to storybook 7 beta and this add on doesn't work
